### ISSUE TITLE
issue with write verilog using hierarchical synthesis

### ISSUE THREAD

**Issue by msaligane** [The-OpenROAD-Project/Resizer#13]
Hi James,

We are trying to use hierarchical synthesis this time on our testcases. 
As a background information, verilog dump using the flatten version is correct . The new version using hierarchical design error seems to come from the resizer writing the verilog:
example:
.links_sif_i\[0\]({link_in_22[1], --> backslashes 

I am attaching the testcase (verilog, reports).
Thanks!

[TESTCASES.zip](https://github.com/The-OpenROAD-Project/Resizer/files/3741679/TESTCASES.zip)

**Comment by jjcherry56** [comment]
There is no coherent description of what the problem is.
What file went into what tool to produce what output that is
produced what error in what other tool?

**Comment by msaligane** [comment]
Hi Minsoo,

Could you please help James on how to locate the issue in the logs? He might not be aware of the alpha release flow. Thanks!

**Comment by msaligane** [comment]
Sorry about that. Hopefully, this will add up more clarity. I added up the makefile, the scripts folder, logs etc.. to the tar file
[testcase-hier-vb.tar.gz](https://github.com/The-OpenROAD-Project/Resizer/files/3747222/testcase-hier-vb.tar.gz)
.
Below is the error I am getting: ( you ll find it in the log files, step 2_1_verilog2def.log)
Error: ./results/tsmc65lp/bsg_manycore_tile/1_synth.v, line 571 syntax error, unexpected ID, expecting '(' or '['.
Error: ./results/tsmc65lp/bsg_manycore_tile/1_synth.v, line 1751 syntax error, unexpected ID, expecting '(' or '['.
Error: ./results/tsmc65lp/bsg_manycore_tile/1_synth.v, line 2539 syntax error, unexpected ID, expecting '(' or '['.
Error: ./results/tsmc65lp/bsg_manycore_tile/1_synth.v, line 3235 syntax error, unexpected ID, expecting '(' or '['.
Error: ./results/tsmc65lp/bsg_manycore_tile/1_synth.v, line 4276 syntax error, unexpected ID, expecting '(' or '['.
Error: ./results/tsmc65lp/bsg_manycore_tile/1_synth.v, line 4518 syntax error, unexpected ID, expecting '(' or '['.
Error: ./results/tsmc65lp/bsg_manycore_tile/1_synth.v, line 4663 syntax error, unexpected ID, expecting '(' or '['.
Error: ./results/tsmc65lp/bsg_manycore_tile/1_synth.v, line 4664 syntax error, unexpected ID, expecting '(' or '['.
Error: ./results/tsmc65lp/bsg_manycore_tile/1_synth.v, line 4667 syntax error, unexpected ID, expecting '(' or '['.
Error: ./results/tsmc65lp/bsg_manycore_tile/1_synth.v, line 4698 syntax error, unexpected ID, expecting '(' or '['.

**Comment by jjcherry56** [comment]
I don’t have internet access right now to download the updated tarfile but I strongly suspect this issue was fixed in opensta (used by verilog2def) with commit 02296d2 on 8/13. So the next question whose answer should be in every bug you report you submit is exactly what code are you running?

**Comment by msaligane** [comment]
I am using the latest version of the alpha release:  v2019-09-05_20-10
https://github.com/The-OpenROAD-Project/alpha-release/releases

**Comment by jjcherry56** [comment]
That doesn't tell me squat about the resizer version.
Run it and it will report the version in the first line of output, like
this:

Resizer 1.2.1 9c78ee0f73 Copyright (c) 2019, Parallax Software, Inc.

**Comment by msaligane** [comment]
Resizer 1.2.1 55264eb826 Copyright (c) 2019, Parallax Software, Inc.
License GPLv3: GNU GPL version 3 <http://gnu.org/licenses/gpl.html>

Let me know if this is an up-to-date version. I have another issue with the def at the floorplan step.  I ll enter that when I have time.

**Comment by jjcherry56** [comment]
I still have no idea what you are trying to report. All I can tell from this error message:

Below is the error I am getting: ( you ll find it in the log files, step 2_1_verilog2def.log)
Error: ./results/tsmc65lp/bsg_manycore_tile/1_synth.v, line 571 syntax error, unexpected ID, expecting '(' or '['.

Is that you are reading the file 1_synth.v into either the resizer or verilogToDef and it contains illegal verilog syntax. That is not a bug in the resizer. It is a bad input to the resizer. Maybe you produced 1_synth.v using verilogToDef but I have no idea how you did that.

I found the file in the directory vb_verilog of the ‘testcase’ you sent. I will repeat myself. You need to describe what file is going into what tool and what options you used to run it.

**Comment by msaligane** [comment]
Great! At least now, we are certain it is not a tool version issue...
You can find the details in the makefile in the latest tar file I shared. It seems it is not clear enough though. So please let me check with Tom and get back to you.

**Comment by msaligane** [comment]
Following is some more details on this issue:
-if you look at the makefile starting at this line: https://github.com/The-OpenROAD-Project/alpha-release/blob/7c0421f6a4cb7f9c5df2fee126b2d40ed72d3d95/flow/Makefile#L114

I am synthesizing the design with no issue till the floorplan step. However, during the floorplan I get the previous error:
_step 2_1_verilog2def.log)
Error: ./results/tsmc65lp/bsg_manycore_tile/1_synth.v, line 571 syntax error, unexpected ID, expecting '('_

Now if I bypass the resizer step:
https://github.com/The-OpenROAD-Project/alpha-release/blob/7c0421f6a4cb7f9c5df2fee126b2d40ed72d3d95/flow/Makefile#L136

and I only provide the verilog out of yosys/abc directly to the next step (floorplan), I do not get that error anymore. So I concluded that the error is coming from the resizer but I might be wrong or not very accurate. The verilog error in question out of the resizer is the one I quoted in my 1st message: 
__.links_sif_i[0]({link_in_22[1], --> backslashes__

I hope this help in identifying the issue. All the files are either on github or in the testcase I shared.

Thanks!
Mehdi
